test/integration-test/TestIIRDeserializer.cpp

```diff
@@ -0,0 +1,598 @@
+        for(int doMethodidx = 0, doMethodSize = lhsStage->getChildren().size();
+            doMethodidx < doMethodSize; ++doMethodidx) {
+          const auto& lhsDoMethod = lhsStage->getChild(doMethodidx);
+          const auto& rhsDoMethod = rhsStage->getChild(doMethodidx);
+          ASSERT_TRUE(lhsDoMethod->getID() == rhsDoMethod->getID());
+          ASSERT_TRUE(lhsDoMethod->getInterval() == rhsDoMethod->getInterval());
+
+          // checking each of the StmtAccesspairs
+          ASSERT_TRUE(lhsDoMethod->getChildren().size() == rhsDoMethod->getChildren().size());
+          for(int stmtidx = 0, stmtSize = lhsDoMethod->getChildren().size(); stmtidx < stmtSize;
+              ++stmtidx) {
+            const auto& lhsStmt = lhsDoMethod->getChild(stmtidx);
+            const auto& rhsStmt = rhsDoMethod->getChild(stmtidx);
+            // check the statement
+            ASSERT_TRUE(lhsStmt->getStatement()->ASTStmt->equals(rhsStmt->getStatement()->ASTStmt.get()));
+
+            // check the accesses
+            ASSERT_TRUE(lhsStmt->getCallerAccesses()->getReadAccesses()
+              == rhsStmt->getCallerAccesses()->getReadAccesses());
+            ASSERT_TRUE(lhsStmt->getCallerAccesses()->getWriteAccesses()
+              == rhsStmt->getCallerAccesses()->getWriteAccesses());
+
+            // if(lhsStmt->getCallerAccesses()) {
+            //   for(const auto& lhsPair : rhsStmt->getCallerAccesses()->getReadAccesses()) {
+            //     ASSERT_TRUE(
+            //         rhsStmt->getCallerAccesses()->getReadAccesses().count(lhsPair.first));
+            //     auto rhsValue = rhsStmt->getCallerAccesses()->getReadAccesses().at(lhsPair.first);
+            //     ASSERT_TRUE(rhsValue == lhsPair.second);
+            //   }
+            //   for(const auto& lhsPair : rhsStmt->getCallerAccesses()->getWriteAccesses()) {
+            //     ASSERT_TRUE(
+            //         rhsStmt->getCallerAccesses()->getWriteAccesses().count(lhsPair.first));
+            //     auto rhsValue = rhsStmt->getCallerAccesses()->getWriteAccesses().at(lhsPair.first);
+            //     ASSERT_TRUE(rhsValue == lhsPair.second);
+            //   }
+            // }
+          }
+        }
+      }
+    }
+  }
+  const auto& lhsControlFlowStmts = lhs->getControlFlowDescriptor().getStatements();
+  const auto& rhsControlFlowStmts = rhs->getControlFlowDescriptor().getStatements();
+
+  ASSERT_TRUE(lhsControlFlowStmts.size() == rhsControlFlowStmts.size());
+  for(int i = 0, size = lhsControlFlowStmts.size(); i < size; ++i) {
+    if(!lhsControlFlowStmts[i]->ASTStmt->equals(rhsControlFlowStmts[i]->ASTStmt.get()))
+      FAIL();
+    if(lhsControlFlowStmts[i]->StackTrace) {
+      if(rhsControlFlowStmts[i]->StackTrace) {
+        for(int j = 0, jsize = lhsControlFlowStmts[i]->StackTrace->size(); j < jsize; ++j) {
+          if(!(lhsControlFlowStmts[i]->StackTrace->at(j) ==
+               rhsControlFlowStmts[i]->StackTrace->at(j))) {
+            FAIL();
```

Comment:
I don't think that you need any of this `FAIL` because all of them are inside a `if` which is the `ASSERT_*` or `EXPECT_*`. Like here, it is
```
ASSERT_EQ(lhsControlFlowStmts[i]->StackTrace->at(j), rhs...);
```
 Eventhough `lhsControlFlowStmts[i]->StackTrace->at(j)` is a `ast::StencilCall*`  which is not comparable I think.